src/main/core/work/event.rs

```diff
@@ -8,51 +9,53 @@ use super::task::TaskRef;
 #[derive(Debug)]
 pub struct Event {
     magic: Magic<Self>,
-    task: TaskRef,
     time: EmulatedTime,
-    src_host_id: HostId,
-    dst_host_id: HostId,
-    src_host_event_id: u64,
+    data: EventData,
     _counter: ObjectCounter,
 }
 
 impl Event {
-    pub fn new(task: TaskRef, time: EmulatedTime, src_host: &Host, dst_host_id: HostId) -> Self {
+    /// A new packet event.
+    pub fn new_packet(packet: Packet, time: EmulatedTime, src_host: &Host) -> Self {
         Self {
             magic: Magic::new(),
-            task,
             time,
-            src_host_id: src_host.id(),
-            dst_host_id,
-            src_host_event_id: src_host.get_new_event_id(),
+            data: EventData::Packet(PacketEventData {
+                packet,
+                src_host_id: src_host.id(),
+                src_host_event_id: src_host.get_new_event_id(),
+            }),
             _counter: ObjectCounter::new("Event"),
         }
     }
 
-    pub fn execute(self, host: &Host) {
-        self.magic.debug_check();
-
-        // make sure we're executing on the correct host
-        assert_eq!(self.host_id(), host.id());
-
-        host.continue_execution_timer();
-        self.task.execute(host);
-        host.stop_execution_timer();
+    /// A new local event.
```

Comment:
Expand the comment a bit to explain what a local event is.

Reply:
Added.

> A new local event, which is an event that was generated locally by the host itself (timers, localhost packets, etc).